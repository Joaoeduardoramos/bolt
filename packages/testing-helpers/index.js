export { isConnected } from './is-connected';
import {
  render as renderServer,
  renderString as renderStringServer,
  stop as stopServer,
} from '@bolt/twig-renderer';
export { fixture as html } from '@open-wc/testing-helpers';

export async function render(template, data) {
  return await renderServer(template, data, true);
}

export async function renderString(template, data) {
  return await renderStringServer(template, data, false);
}

export { stopServer };

// import {
//   // isConnected,
//   // render,
//   // renderString,
//   // stopServer,
//   html,
// } from '../../../testing-helpers';
