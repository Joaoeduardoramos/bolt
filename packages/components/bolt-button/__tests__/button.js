import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';

const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../button.schema.yml'));
const { tag } = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 60000;

describe('button', async () => {
  let page;

  afterAll(async () => {
    await stopTwigRenderer();
  }, timeout);

  beforeEach(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
      waitLoad: true,
      waitNetworkIdle: true, // defaults to false
    });
  }, timeout);

  test('Basic usage', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'This is a button',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  tag.enum.forEach(async tagChoice => {
    test(`Button tag: ${tagChoice}`, async () => {
      const results = await renderTwig('@bolt-components-button/button.twig', {
        text: 'This is a button',
        tag: tagChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test('Button with outer classes via Drupal Attributes', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button with outer classes',
      attributes: {
        class: ['u-bolt-padding-medium'],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Button with inner classes via Drupal Attributes', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button with inner classes',
      attributes: {
        class: ['is-active'],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Button with outer JS-class via Drupal Attributes', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button with outer JS-prefixed class',
      attributes: {
        class: ['js-click-me'],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Button with c-bolt- class is thrown out', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button with outer JS-prefixed class',
      attributes: {
        class: ['c-bolt-button--secondary'],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Button with an onClick param renders properly', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button with onClick via param',
      onClick: 'on-click-test',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Button with an onClick attributes renders properly', async () => {
    const results = await renderTwig('@bolt-components-button/button.twig', {
      text: 'Button w/ onClick via attributes',
      attributes: {
        'on-click': 'on-click-test',
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-button> w/o Shadow DOM renders', async function() {
    const renderedButtonHTML = await page.evaluate(() => {
      const btn = document.createElement('bolt-button');
      btn.textContent = 'This is a button';
      document.body.appendChild(btn);
      btn.useShadow = false;
      btn.updated();
      return btn.outerHTML;
    });

    const renderedHTML = await html(renderedButtonHTML);

    expect(
      renderedHTML
        .querySelector('.c-bolt-button')
        .classList.contains('c-bolt-button--primary'),
    ).toBe(true);

    const image = await page.screenshot();
    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-button> with Shadow DOM renders', async function() {
    const defaultButtonShadowRoot = await page.evaluate(() => {
      const btn = document.createElement('bolt-button');
      btn.textContent = 'Button Test -- Shadow Root HTML';
      document.body.appendChild(btn);
      btn.updated();
      return btn.renderRoot.innerHTML;
    });

    const defaultButtonOuter = await page.evaluate(() => {
      const btn = document.createElement('bolt-button');
      btn.textContent = 'Button Test -- Outer HTML';
      document.body.appendChild(btn);
      btn.updated();
      return btn.outerHTML;
    });

    const renderedShadowDomHTML = await html(defaultButtonShadowRoot);
    const renderedHTML = await html(defaultButtonOuter);

    expect(renderedHTML.textContent).toEqual('Button Test -- Outer HTML');
    // expect(
    //   renderedShadowDomHTML
    //     .querySelector('.c-bolt-button')
    //     .classList.contains('c-bolt-button--primary'),
    // ).toBe(true);

    // expect(renderedShadowDomHTML.querySelector('style')).toBe(true);
    // expect(renderedShadowDomHTML.querySelector('button')).toBe(true);

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.01',
      failureThresholdType: 'percent',
    });

    expect(renderedShadowDomHTML).toMatchSnapshot();
    expect(renderedHTML).toMatchSnapshot();
  });
});
