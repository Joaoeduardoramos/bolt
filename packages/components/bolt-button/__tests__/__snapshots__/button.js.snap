// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button Button with an onClick attributes renders properly 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
             on-click="on-click-test"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button w/ onClick via attributes
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Button with an onClick param renders properly 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
             on-click="on-click-test"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button with onClick via param
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Button with c-bolt- class is thrown out 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button with outer JS-prefixed class
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Button with inner classes via Drupal Attributes 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center is-active"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button with inner classes
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Button with outer JS-class via Drupal Attributes 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
             class="js-click-me"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button with outer JS-prefixed class
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Button with outer classes via Drupal Attributes 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
             class="u-bolt-padding-medium"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Button with outer classes
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button Default <bolt-button> w/o Shadow DOM renders 2`] = `
<bolt-button
  style=""
>
  <!---->
  
       
        
  <button
    class="c-bolt-button c-bolt-button--center c-bolt-button--primary c-bolt-button--medium c-bolt-button--border-radius-regular"
    style=""
  >
    
          
    <!---->
    
                
    <span
      class="c-bolt-button__icon is-empty"
      style=""
    >
      <!---->
      
                        
      <slot
        name="before"
        style=""
      >
        
                      
      </slot>
      <!---->
    </span>
    
              
    <!---->
    
                
    <span
      class="c-bolt-button__item"
      style=""
    >
      <!---->
      <!---->
      <!---->
      Hello World!
      <!---->
      <!---->
      <!---->
    </span>
    
              
    <!---->
    
                
    <span
      class="c-bolt-button__icon is-empty"
      style=""
    >
      <!---->
      
                        
      <slot
        name="after"
        style=""
      >
        
                      
      </slot>
      <!---->
    </span>
    
              
    <!---->
    
        
  </button>
  
      
    
  <!---->
</bolt-button>
`;

exports[`button Default <bolt-button> with Shadow DOM renders 2`] = `
<style
  style=""
>
  
            bolt-button{display:inline-block;display:inline-flex}.c-bolt-button{padding:0;font-family:-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",sans-serif;font-family:var(--bolt-font-family-body);-webkit-appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;font-weight:600;box-shadow:0 1px 4px 1px rgba(6,10,36,.1),0 5px 10px 0 rgba(6,10,36,.08);transition:.3s cubic-bezier(.25,.8,.25,1);display:inline-block;display:inline-flex;flex-direction:row;flex-grow:1;align-items:center;position:relative;-webkit-transform:translate3d(0,0,0);transform:translate3d(0,0,0);text-decoration:none;vertical-align:middle;cursor:pointer;border-radius:3px;border-width:1px;border-style:solid}.js-fonts-loaded .c-bolt-button{font-family:"Open Sans","Helvetica Neue",sans-serif;font-family:var(--bolt-font-family-body)}.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]).is-hover,.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]):hover{-webkit-transform:translateY(-2px);transform:translateY(-2px);box-shadow:0 1px 8px 1px rgba(6,10,36,.18),0 5px 10px 1px rgba(6,10,36,.15),0 15px 30px 0 rgba(6,10,36,.16)}.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]).is-hover:before,.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]):hover:before{opacity:1}.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]).is-active,.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]):active{-webkit-transform:translate3d(0,1px,0);transform:translate3d(0,1px,0)}.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]).is-active:before,.c-bolt-button:not(.c-bolt-button--disabled):not(.c-bolt-button--text):not([disabled]):active:before{opacity:0}.c-bolt-button:before{content:'';display:block;opacity:.2;position:absolute;top:0;left:0;z-index:-5;width:100%;height:100%;pointer-events:none;border-radius:3px;-webkit-transform:translateY(-2px);transform:translateY(-2px)}.c-bolt-button--primary,.c-bolt-button--primary:visited{color:rgba(var(--bolt-theme-text-on-primary), 1);border-color:rgba(var(--bolt-theme-primary), 1);background-color:rgba(var(--bolt-theme-primary), 1)}.c-bolt-button--primary.is-hover,.c-bolt-button--primary:hover{color:rgba(var(--bolt-theme-text-on-primary-lighten-15), 1);border-color:rgba(var(--bolt-theme-primary-lighten-15), 1);background-color:rgba(var(--bolt-theme-primary-lighten-15), 1)}.c-bolt-button--primary.is-focus,.c-bolt-button--primary:focus{color:rgba(var(--bolt-theme-text-on-primary-darken-15), 1);border-color:rgba(var(--bolt-theme-primary-darken-15), 1);background-color:rgba(var(--bolt-theme-primary-darken-15), 1)}.c-bolt-button--primary.is-active,.c-bolt-button--primary:active{color:rgba(var(--bolt-theme-text-on-primary-darken-25), 1);border-color:rgba(var(--bolt-theme-primary-darken-25), 1);background-color:rgba(var(--bolt-theme-primary-darken-25), 1)}.c-bolt-button--secondary,.c-bolt-button--secondary:visited{color:rgba(var(--bolt-theme-text-on-secondary), 1);border-color:rgba(var(--bolt-theme-secondary), 1);background-color:rgba(var(--bolt-theme-secondary), 1)}.c-bolt-button--secondary.is-hover,.c-bolt-button--secondary:hover{color:rgba(var(--bolt-theme-text-on-secondary-lighten-5), 1);border-color:rgba(var(--bolt-theme-secondary-lighten-5), 1);background-color:rgba(var(--bolt-theme-secondary-lighten-5), 1)}.c-bolt-button--secondary.is-focus,.c-bolt-button--secondary:focus{color:rgba(var(--bolt-theme-text-on-secondary-darken-3), 1);border-color:rgba(var(--bolt-theme-secondary-darken-3), 1);background-color:rgba(var(--bolt-theme-secondary-darken-3), 1)}.c-bolt-button--secondary.is-active,.c-bolt-button--secondary:active{color:rgba(var(--bolt-theme-text-on-secondary-darken-10), 1);border-color:rgba(var(--bolt-theme-secondary-darken-10), 1);background-color:rgba(var(--bolt-theme-secondary-darken-10), 1)}.c-bolt-button--text{text-decoration:none;border-color:transparent;background-color:transparent;opacity:1;color:rgba(var(--bolt-theme-headline-link), 1)}.c-bolt-button--text,.c-bolt-button--text:before{border-radius:0;box-shadow:none}.c-bolt-button--text:visited{opacity:1}.c-bolt-button--text:hover:not([disabled]){opacity:.8}.c-bolt-button--text:active,.c-bolt-button--text:focus:active{opacity:.6}.c-bolt-button--disabled,.c-bolt-button[disabled]{cursor:not-allowed}.c-bolt-button--disabled,.c-bolt-button--disabled:active,.c-bolt-button--disabled:hover,.c-bolt-button--disabled:visited,.c-bolt-button[disabled],.c-bolt-button[disabled]:active,.c-bolt-button[disabled]:hover,.c-bolt-button[disabled]:visited{color:rgba(var(--bolt-theme-text-disabled), 1);border-color:rgba(var(--bolt-theme-disabled), 1);background-color:rgba(var(--bolt-theme-disabled), 1)}.c-bolt-button--disabled:active:before,.c-bolt-button--disabled:before,.c-bolt-button--disabled:hover:before,.c-bolt-button[disabled]:active:before,.c-bolt-button[disabled]:before,.c-bolt-button[disabled]:hover:before{opacity:1}.c-bolt-button--disabled,.c-bolt-button--disabled:active,.c-bolt-button--disabled:hover,.c-bolt-button[disabled],.c-bolt-button[disabled]:active,.c-bolt-button[disabled]:hover{-webkit-transform:none;transform:none}.c-bolt-button--uppercase{text-transform:uppercase}.c-bolt-button--lowercase{text-transform:lowercase}.c-bolt-button--capitalize{text-transform:capitalize}.c-bolt-button--medium{padding:.825rem 2rem;font-size:.8rem;line-height:1.45}.c-bolt-button--medium.c-bolt-button--icon-only{padding:2rem}.c-bolt-button--xxsmall{padding:.103rem .25rem;font-size:.8rem;line-height:1.45}.c-bolt-button--xxsmall.c-bolt-button--icon-only{padding:.25rem}.c-bolt-button--xsmall{padding:.206rem .5rem;font-size:.8rem;line-height:1.45}.c-bolt-button--xsmall.c-bolt-button--icon-only{padding:.5rem}.c-bolt-button--small{padding:.412rem 1rem;font-size:.8rem;line-height:1.45}.c-bolt-button--small.c-bolt-button--icon-only{padding:1rem}.c-bolt-button--large{padding:.825rem 2rem;font-size:1rem;line-height:1.65}.c-bolt-button--large.c-bolt-button--icon-only{padding:2rem}.c-bolt-button--xlarge{padding:1.65rem 4rem;font-size:1.111rem;line-height:1.45}.c-bolt-button--xlarge.c-bolt-button--icon-only{padding:4rem}.c-bolt-button--full,bolt-button[width=full]{width:100%}:host([width=full]){width:100%}@media (min-width:37.5em){.c-bolt-button--full\\@small,bolt-button[width='full@small']{width:100%}:host([width='full@small']){width:100%}}.c-bolt-button--border-radius-regular,.c-bolt-button--border-radius-regular:before{border-radius:3px}.c-bolt-button--border-radius-full,.c-bolt-button--border-radius-full:before{border-radius:50em}.c-bolt-button--center{justify-content:center;text-align:center}.c-bolt-button--start{justify-content:flex-start;text-align:left;text-align:start}.c-bolt-button--end{justify-content:flex-end;text-align:right;text-align:end}.c-bolt-button__item+.c-bolt-button__icon:not(.is-empty){margin-left:.25rem}.c-bolt-button--icon-only .c-bolt-button__item+.c-bolt-button__icon:not(.is-empty){margin-left:0}.c-bolt-button__icon:not(.is-empty)+.c-bolt-button__item{margin-left:.5rem}.c-bolt-button--icon-only .c-bolt-button__icon:not(.is-empty)+.c-bolt-button__item{margin-left:0}.c-bolt-button__icon{display:inline-block;display:inline-flex;align-self:center;line-height:1;text-align:center;vertical-align:middle;transition:-webkit-transform 150ms ease-in-out;transition:transform 150ms ease-in-out;transition:transform 150ms ease-in-out, -webkit-transform 150ms ease-in-out} .u-bolt-visuallyhidden{position:absolute!important;width:1px!important;height:1px!important;overflow:hidden!important;margin:-1px!important;padding:0!important;border:0!important;clip:rect(0 0 0 0)!important;-webkit-clip-path:inset(50%)!important;clip-path:inset(50%)!important;white-space:nowrap!important}
          
</style>
`;

exports[`button Default <bolt-button> with Shadow DOM renders 3`] = `
<bolt-button
  style=""
>
  Button Test -- Outer HTML
</bolt-button>
`;

exports[`button Twig-integrated server-side rendering can pre-render the <bolt-button> web component 1`] = `

<bolt-button>
  <button class="c-bolt-button c-bolt-button--center c-bolt-button--primary c-bolt-button--medium c-bolt-button--border-radius-regular">
    <span class="c-bolt-button__icon is-empty">
      <slot name="before">
      </slot>
    </span>
    <span class="c-bolt-button__item">
      SSR via pure web component!
      <bolt-icon name="chevron-right">
        <div class="c-bolt-icon c-bolt-icon--chevron-right">
          <svg class="c-bolt-icon__icon"
               viewbox="0 0 32 32"
          >
            <path fill="var(--bolt-theme-icon, currentColor)"
                  fill-rule="evenodd"
                  d="M12.938 7.054a1.34 1.34 0 0 0-1.884 0 1.339 1.339 0 0 0 0 1.884l7.059 7.06-7.059 7.059a1.34 1.34 0 0 0 0 1.884 1.329 1.329 0 0 0 1.883 0l8-8a1.323 1.323 0 0 0 0-1.885l-7.999-8.002z"
            >
            </path>
          </svg>
        </div>
      </bolt-icon>
    </span>
    <span class="c-bolt-button__icon is-empty">
      <slot name="after">
      </slot>
    </span>
  </button>
</bolt-button>

`;

exports[`button basic button 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button button tag: a 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <a class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
     is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </a>
</bolt-button>

`;

exports[`button button tag: button 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button button tag: link 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <a class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
     is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </a>
</bolt-button>

`;

exports[`button button tag: reset 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button type="reset"
          class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;

exports[`button button tag: submit 1`] = `

<bolt-button align="center"
             color="primary"
             size="medium"
             transform="none"
             border-radius="regular"
>
  <button type="submit"
          class="c-bolt-button c-bolt-button--medium c-bolt-button--border-radius-regular c-bolt-button--primary c-bolt-button--center"
          is="shadow-root"
  >
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
    <replace-with-children class="c-bolt-button__item ">
      Hello World
    </replace-with-children>
    <replace-with-children class="c-bolt-button__icon is-empty">
    </replace-with-children>
  </button>
</bolt-button>

`;
